library/src/blas1/rocblas_scal_batched.hpp

```diff
@@ -0,0 +1,92 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#ifndef __SCAL_BATCHED_HPP__
+#define __SCAL_BATCHED_HPP__
+
+#include "handle.h"
+#include "logging.h"
+#include "rocblas.h"
+#include "scal_host.hpp"
+#include "utility.h"
+
+template <typename T, typename = T>
+constexpr char rocblas_scal_name[] = "unknown";
+template <>
+constexpr char rocblas_scal_name<float>[] = "rocblas_sscal_batched";
+template <>
+constexpr char rocblas_scal_name<double>[] = "rocblas_dscal_batched";
+template <>
+constexpr char rocblas_scal_name<rocblas_float_complex>[] = "rocblas_cscal_batched";
+template <>
+constexpr char rocblas_scal_name<rocblas_double_complex>[] = "rocblas_zscal_batched";
+template <>
+constexpr char rocblas_scal_name<rocblas_float_complex, float>[] = "rocblas_csscal_batched";
+template <>
+constexpr char rocblas_scal_name<rocblas_double_complex, double>[] = "rocblas_zdscal_batched";
+
+template <typename T, typename U>
+rocblas_status rocblas_scal_batched_impl(rocblas_handle handle,
+                                            rocblas_int    n,
+                                            const U*       alpha,
+                                            T*             x[],
+                                            rocblas_int    offsetx,
```

Comment:
For my understanding, what is the purpose of this offsetx parameter? Seems to be an implementation convenience but I don't see the immediate benefit of it.

Reply:
As we discussed earlier, the offset (or shift) parameter can be used in batched calls internally to use only subsections of data in each batch. This can be done manually in strided_batched and regular calls, but not as easily in batched calls. I'm not sure of any current use for functions such as scal, but if you look at trsm, it uses offsets when it makes calls to gemm.

Reply:
Why not just pass a pointer to the offseted data, rather than a pointer and offset?

Reply:
The batched version uses an array of pointers (for each batch). The offset parameter will offset each batch after being called, rather than having to create an array to the offset data for each batch.